src/Engine/Spells/CastSpellInfo.cpp

```diff
@@ -2754,51 +2803,47 @@ void CastSpellInfoHelpers::CastSpell() {
-                            GameTime(pParty->GetPlayingTime() + GameTime::FromSeconds(spellduration)),
+                            pParty->GetPlayingTime() + spell_duration,
                             spell_mastery, amount, 0, 0);
                     pParty->pPartyBuffs[PARTY_BUFF_RESIST_FIRE].Apply(
-                            GameTime(pParty->GetPlayingTime() + GameTime::FromSeconds(spellduration)),
+                            pParty->GetPlayingTime() + spell_duration,
                             spell_mastery, amount, 0, 0);
                     pParty->pPartyBuffs[PARTY_BUFF_RESIST_WATER].Apply(
-                            GameTime(pParty->GetPlayingTime() + GameTime::FromSeconds(spellduration)),
+                            pParty->GetPlayingTime() + spell_duration,
                             spell_mastery, amount, 0, 0);
                     pParty->pPartyBuffs[PARTY_BUFF_RESIST_AIR].Apply(
-                            GameTime(pParty->GetPlayingTime() + GameTime::FromSeconds(spellduration)),
+                            pParty->GetPlayingTime() + spell_duration,
                             spell_mastery, amount, 0, 0);
                     pParty->pPartyBuffs[PARTY_BUFF_RESIST_EARTH].Apply(
-                            GameTime(pParty->GetPlayingTime() + GameTime::FromSeconds(spellduration)),
+                            pParty->GetPlayingTime() + spell_duration,
                             spell_mastery, amount, 0, 0);
                     pParty->pPartyBuffs[PARTY_BUFF_FEATHER_FALL].Apply(
-                            GameTime(pParty->GetPlayingTime() + GameTime::FromSeconds(spellduration)),
+                            pParty->GetPlayingTime() + spell_duration,
                             spell_mastery, spell_level + 5, 0, 0);
                     pParty->pPartyBuffs[PARTY_BUFF_WIZARD_EYE].Apply(
-                            GameTime(pParty->GetPlayingTime() + GameTime::FromSeconds(spellduration)),
+                            pParty->GetPlayingTime() + spell_duration,
                             spell_mastery, spell_level + 5, 0, 0);
                     break;
                 }
 
                 case SPELL_LIGHT_HOUR_OF_POWER:
                 {
+                    GameTime haste_duration, other_duration;
+
                     switch (spell_mastery) {
-                        case PLAYER_SKILL_MASTERY_NOVICE:
-                            spellduration = 4;
-                            amount = 4;
-                            break;
-                        case PLAYER_SKILL_MASTERY_EXPERT:
-                            spellduration = 4;
-                            amount = 4;
-                            break;
                         case PLAYER_SKILL_MASTERY_MASTER:
-                            spellduration = 12;
-                            amount = 12;
+                            haste_duration = GameTime::FromHours(1).AddMinutes(3 * spell_level * 5);
```

Comment:
You've changed the duration from 1h + 12m/skill to 1h + 15m/skill.

If this was intentional then please add a doc, e.g. `// was 1h + 12min/skill point`